zap/src/irgen/mod.rs

```diff
@@ -441,6 +442,15 @@ impl Display for Expr {
 			Self::True => write!(f, "true"),
 			Self::Nil => write!(f, "nil"),
 
+			Self::StrOrBool(string) => {
+				if string == "false" || string == "False" {
```

Comment:
I think this is a gotcha. Let's make it more explicit so that people can still use `False` if they want a string.
Or, we can put this behind a setting. Up to you.

If you go with the first option:
```suggestion
				if string == "false" {
```